src/image/Image.cc

```diff
@@ -697,6 +699,30 @@ geom::Box2I bboxFromMetadata(daf::base::PropertySet & metadata)
     return geom::Box2I(xy0, dims);
 }
 
+namespace {
+
+template <typename A, typename B, typename PtrLess = std::less<void const*>>
+bool ptrsOverlap(A beg1Addr, A end1Addr, B beg2Addr, B end2Addr, PtrLess ptrLess = PtrLess()) {
+    return ptrLess(beg1Addr, end2Addr) && ptrLess(beg2Addr, end1Addr);
+}
```

Comment:
I think this could be simplified a bit:

    bool ptrsOverlap(void const* const beg1Addr, void const* const end1Addr, void const* const beg2Addr, void const* const end2Addr) {
        static std::less<void const*> ptrLess;    // std::less is safer than < for generic pointers
        return ptrLess(beg1Addr, end2Addr) && ptrLess(beg2Addr, end1Addr);
    }

(depending on how rigorous you want to get with `const`, of course).